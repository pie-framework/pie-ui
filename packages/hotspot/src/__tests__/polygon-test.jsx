import React from 'react';
import _ from 'lodash';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Konva from 'konva';

import Polygon from '../hotspot/polygon';

Konva.isBrowser = false;

describe('Polygon', () => {
  let onClick, wrapper;

  const mkWrapper = (opts = {}) => {
    opts = _.extend(
      {
        classes: {
          base: 'base'
        },
        height: 200,
        hotspotColor: 'rgba(137, 183, 244, 0.65)',
        id: '1',
        isCorrect: false,
        isEvaluateMode: false,
        disabled: false,
        outlineColor: 'blue',
        selected: false,
        points: [{ x: 94, y: 4 }, { x: 89, y: 4 }, { x: 36, y: 40 }]
      },
      opts
    );

    return shallow(<Polygon {...opts} onClick={onClick} />);
  };

  beforeEach(() => {
    onClick = jest.fn();
    wrapper = mkWrapper();
  });

  describe('snapshots', () => {
    describe('outline color', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ outlineColor: 'red' });
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });

    describe('hotspot color', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ hotspotColor: 'rgba(217, 30, 24, 0.65)' });
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });

    describe('evaluate with correct answer', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ isEvaluateMode: true, isCorrect: true });
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
