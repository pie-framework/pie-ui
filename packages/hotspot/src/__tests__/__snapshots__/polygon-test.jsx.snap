// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Polygon snapshots evaluate with correct answer renders 1`] = `
<PolygonComponent
  classes={
    Object {
      "base": "PolygonComponent-base-1 base",
    }
  }
  disabled={false}
  height={200}
  hotspotColor="rgba(137, 183, 244, 0.65)"
  id="1"
  isCorrect={true}
  isEvaluateMode={true}
  onClick={[MockFunction]}
  outlineColor="blue"
  points={
    Array [
      Object {
        "x": 94,
        "y": 4,
      },
      Object {
        "x": 89,
        "y": 4,
      },
      Object {
        "x": 36,
        "y": 40,
      },
    ]
  }
  selected={false}
/>
`;

exports[`Polygon snapshots hotspot color renders 1`] = `
<PolygonComponent
  classes={
    Object {
      "base": "PolygonComponent-base-1 base",
    }
  }
  disabled={false}
  height={200}
  hotspotColor="rgba(217, 30, 24, 0.65)"
  id="1"
  isCorrect={false}
  isEvaluateMode={false}
  onClick={[MockFunction]}
  outlineColor="blue"
  points={
    Array [
      Object {
        "x": 94,
        "y": 4,
      },
      Object {
        "x": 89,
        "y": 4,
      },
      Object {
        "x": 36,
        "y": 40,
      },
    ]
  }
  selected={false}
/>
`;

exports[`Polygon snapshots outline color renders 1`] = `
<PolygonComponent
  classes={
    Object {
      "base": "PolygonComponent-base-1 base",
    }
  }
  disabled={false}
  height={200}
  hotspotColor="rgba(137, 183, 244, 0.65)"
  id="1"
  isCorrect={false}
  isEvaluateMode={false}
  onClick={[MockFunction]}
  outlineColor="red"
  points={
    Array [
      Object {
        "x": 94,
        "y": 4,
      },
      Object {
        "x": 89,
        "y": 4,
      },
      Object {
        "x": 36,
        "y": 40,
      },
    ]
  }
  selected={false}
/>
`;
